### PR TITLE
Feature/pools controller

### PR DESCRIPTION
**Pull request by rripken**
Initial implementation of the pools controller and dao.  I don't love the way pagination works but maybe we can get someone to revisit the pl/sql for that and provide better procedures

### CODE REVIEW

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+import usace.cwms.db.jooq.codegen.tables.AV_CWMS_TS_ID2;
+import usace.cwms.db.jooq.codegen.tables.AV_TSV;
+import usace.cwms.db.jooq.codegen.tables.AV_TSV_DQU;
+
+import static org.jooq.impl.DSL.asterisk;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.max;
+import static org.jooq.impl.DSL.partitionBy;
+
+public class RecentValueDao extends JooqDao<TsvDqu>
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java]: This is clearly just timeseries operations, albeit slightly specific ones, does it really make sense to have a separate DAO or should this all just be lumped into the/a TimeSeriesDAO? (I'm not actually sure, just trying to start a discussion.)

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	{
+		super(p.getPoolName(), p.getProjectId(), p.getBottomLevelId(), p.getTopLevelId(), p.isImplicit());
+	}
+
+	public Number getAttribute()
+	{
+		return attribute;
+	}
+
+	public void setAttribute(Number attribute)
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java]: The DTOs should be read only, use the builder pattern if everything can't be handled in the constructor.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/Tsv.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
+package cwms.radar.data.dto;
+
+import java.util.Date;
+import javax.persistence.EmbeddedId;
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/Tsv.java]: What led to requiring these imports?

**Review by MikeNeilson — DISMISSED** [review]

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java (hunk 1 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java]: I just pulled the TimeSeries methods out of CwmsDataManager and combined them with these methods into a TimeSeriesDao object.   These queries were ported over from my cwms mobile queries and I'm not 100% sure I want to expose TsvDqu and Tsv objects but at the moment the recent methods aren't access by the controllers

Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java]: I would say that at the least, that shouldn't be the name. But couldn't disagree with the concept enough to say "no don't do that."

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java (hunk 2 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java]: ok. I reworked Pool and Pools

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/Tsv.java (hunk 3 above)
Comment (rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/Tsv.java]: CWMSMobile used similar classes with hibernate - I should have removed them.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java [hunk 4]
```diff
@@ -8,51 +8,33 @@
...
-	}
-
-	public Pool(PoolType p)
-	{
-		super(p.getPoolName(), p.getProjectId(), p.getBottomLevelId(), p.getTopLevelId(), p.isImplicit());
+	private final Number attribute;
+	private final String description;
+	private final String clobText;
+
+	public Pool(Builder b){
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/Pool.java]: Should this be private?

**Review by MikeNeilson — DISMISSED** [review]
Looks good to me.

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**